worlds/hades/Client.py

```diff
@@ -0,0 +1,424 @@
+        hades_settings_string += str(self.hades_slot_data["keepsakes_needed"]) + "-"
+        hades_settings_string += str(self.hades_slot_data["fates_needed"]) + "-"
+            
+        #Store the compact setting string
+        self.compact_setting_string = hades_settings_string
+        
+
+    def request_location_to_item_dictionary(self):
+        self.creating_location_to_item_mapping = True
+        request = self.missing_locations_cache + self.checked_locations_cache
+        asyncio.create_task(self.send_msgs([{"cmd": "LocationScouts", "locations": request, "create_as_hint": 0}]))
+
+    async def create_location_to_item_dictionary(self, itemsdict):
+        locationItemMapping = ""
+        for networkitem in itemsdict:
+            locationItemMapping += self.clear_invalid_char(self.location_names.lookup_in_slot(networkitem.location)) + "--" \
+            + self.clear_invalid_char(self.player_names[networkitem.player]) + "--" \
+            + self.clear_invalid_char(self.item_names.lookup_in_slot(networkitem.item, networkitem.player)) + "||"
+            
+        subsume.Send(styx_scribe_send_prefix + "Location to Item Map:" + locationItemMapping)
+        self.creating_location_to_item_dictionary = False
+        subsume.Send(styx_scribe_send_prefix + "Data finished"+self.compact_setting_string)
+    
+    def clear_invalid_char(self, inputstring: str):
+        newstr = inputstring.replace("{", "")
+        newstr = newstr.replace("}","")
+        return newstr
+
+    # ----------------- Package Management section ends --------------------------------
+
+    # ----------------- Hints from game section starts --------------------------------
+
+    async def send_location_hint_to_server(self, message):
+        if (self.hades_slot_data["store_give_hints"] == 0):
+            return;
+        split_array = message.split("-")
+        request = []
+        for location in split_array:
+            if (len(location)>0):
+                request.append(self.location_name_to_id[location])
+        asyncio.create_task(self.send_msgs([{"cmd": "LocationScouts", "locations": request, "create_as_hint": 2}]))
+
+    # ----------------- Hints from game section ends --------------------------------
+
+    # -------------deathlink section started --------------------------------
+    def on_deathlink(self, data: dict):
+        # What should be done when a deathlink message is recieved
+        if (self.deathlink_pending):
+            return
+        self.deathlink_pending = True
+        subsume.Send(styx_scribe_send_prefix + "Deathlink recieved")
+        super().on_deathlink(data)
+        asyncio.create_task(self.wait_and_lower_deathlink_flag())
+
+    def send_death(self, death_text: str = ""):
+        # What should be done to send a death link
+        # Avoid sending death if we died from a deathlink
+        if (self.deathlink_enabled == False or self.deathlink_pending == True):
+            return
```

Comment:
When comparing to False or True, it is recommend to use `is` or `is not`.
Alternatively, you could do:
```py
if not self.deathlink_enabled or self.deathlink_pending:
```